src/dataaggregator/aggregator/metricPoller.go

```diff
@@ -26,7 +27,7 @@ type MetricPoller struct {
 func NewMetricPoller(metricCollectorUrl string, logger lager.Logger, appChan chan *appmetric.AppMonitor, metricConsumer MetricConsumer, httpClient *http.Client) *MetricPoller {
 	return &MetricPoller{
 		metricCollectorUrl: metricCollectorUrl,
-		logger:             logger.Session("metric-poller"),
+		logger:             logger,
```

Comment:
Let's be consistent. Just pass the logger through and let the function set the name, so what you had was correct. This will prevent nesting, like aggregator-metric-poller while following a common pattern.